src/Dhall/Import.hs

```diff
@@ -539,26 +539,13 @@ getCacheFile hash = do
 
                     liftIO (Directory.setPermissions directory private)
 
-    let tryXdgCacheHome = do
+    cacheDirectory <- do
```

Comment:
You can also simplify this a bit further by inlining the inner `do` block into the outer `do` block.  In other words, instead of this:

```haskell
cacheDirectory <- do
    cacheDirectory <- liftIO $ Directory.getXdgDirectory Directory.XdgCache ""

    assertDirectory cacheDirectory

    return cacheDirectory
```

... you can do this:

```haskell
cacheDirectory <- liftIO $ Directory.getXdgDirectory Directory.XdgCache ""

assertDirectory cacheDirectory
```

Reply:
yeah, the nesting do is unnecesary